### ISSUE TITLE
Jupyter kernel dies while executing qutip.eigenstate(sparse = True)

### ISSUE THREAD

**Issue by shemian29** [qutip/qutip#1705]
An issue has come when calculating eigenstates with the sparse = True option, which did not happen before. This issue has appeared in two different computers (both Linux), and with fresh installations of qutip.

An example that illustrates the issue is:

  import qutip as qt
  A = qt.tensor(qt.sigmax(),qt.sigmay())
  A.eigenstates(sparse = True, eigvals = 1)

When this runs, the Kernel dies:

![Screenshot from 2021-11-02 17-28-16](https://user-images.githubusercontent.com/48872344/139960518-9708a7f9-5f6f-4be3-a5f0-1db498736d9b.png)

I'm not sure what the source of the issue is. I have used the sparse=True option many times before without any problem. Any suggestions are appreciated.

This is the output of qutip.about() for one of the computers:

QuTiP: Quantum Toolbox in Python
-------------------------------------------------
Copyright (c) QuTiP team 2011 and later.
Current admin team: Alexander Pitchford, Nathan Shammah, Shahnawaz Ahmed, Neill Lambert, Eric Giguère, Boxi Li, Jake Lishman and Simon Cross.
Board members: Daniel Burgarth, Robert Johansson, Anton F. Kockum, Franco Nori and Will Zeng.
Original developers: R. J. Johansson & P. D. Nation.
Previous lead developers: Chris Granade & A. Grimsmo.
Currently developed through wide collaboration. See https://github.com/qutip for details.

QuTiP Version:      4.6.2
Numpy Version:      1.21.2
Scipy Version:      1.7.1
Cython Version:     0.29.24
Matplotlib Version: 3.4.3
Python Version:     3.8.12
Number of CPUs:     8
BLAS Info:          INTEL MKL
OPENMP Installed:   False
INTEL MKL Ext:      True
Platform Info:      Linux (x86_64)
Installation path:  /home/shemian29/anaconda3/envs/Tokyo38/lib/python3.8/site-packages/qutip
-------------------------------------------------
Please cite QuTiP in your publication.
-------------------------------------------------
For your convenience a bibtex reference can be easily generated using `qutip.cite()`

**Comment by hodgestar** [comment]
@shemian29 Did this issue occur after you upgraded QuTiP or one of it's dependencies? And how did you install QuTiP? The IPython kernel crash suggests that maybe the installation is broken somehow (e.g. mismatched C libraries, etc).

I can't reproduce your error on my side.

**Comment by AGaliciaMartinez** [comment]
I can not reproduce it either. Here is _my_ output from `qutip.about()`. I seem to have a different BLAS installation but I am not sure if it is related.

```
QuTiP: Quantum Toolbox in Python
================================
Copyright (c) QuTiP team 2011 and later.
Current admin team: Alexander Pitchford, Nathan Shammah, Shahnawaz Ahmed, Neill Lambert, Eric Giguère, Boxi Li, Jake Lishman and Simon Cross.
Board members: Daniel Burgarth, Robert Johansson, Anton F. Kockum, Franco Nori and Will Zeng.
Original developers: R. J. Johansson & P. D. Nation.
Previous lead developers: Chris Granade & A. Grimsmo.
Currently developed through wide collaboration. See https://github.com/qutip for details.

QuTiP Version:      4.6.2
Numpy Version:      1.21.3
Scipy Version:      1.7.1
Cython Version:     None
Matplotlib Version: None
Python Version:     3.9.5
Number of CPUs:     4
BLAS Info:          OPENBLAS
OPENMP Installed:   False
INTEL MKL Ext:      False
Platform Info:      Linux (x86_64)
Installation path:  /home/kaladin/.virtualenvs/fresh-qutip/lib/python3.9/site-packages/qutip
================================================================================
Please cite QuTiP in your publication.
================================================================================
For your convenience a bibtex reference can be easily generated using `qutip.cite()`
```

**Comment by hodgestar** [comment]
@shemian29 It's possible that the error is caused by using numpy 1.21.2 with MKL which triggers this bug -- https://github.com/numpy/numpy/issues/20356 -- on Intel CPUs that support AVX512.

If that is the case, it should be fixed by install QuTiP 4.6.3 when it is released (hopefully early next week) and numpy 1.22.

**Comment by hodgestar** [comment]
QuTiP 4.6.3 has be released with the likely fix and we have no more information or ability to reproduce, so closing for now.
